packages/api/amplify_api/example/lib/graphql_api_view.dart

```diff
@@ -62,127 +69,549 @@ class _GraphQLApiViewState extends State<GraphQLApiView> {
+    );
+
+    final operation = Amplify.API.mutate(
+      request: req,
+    );
+    _lastOperation = operation;
+
+    final response = await operation.response;
+    handleResponse(response);
   }
 
-  Future<void> mutate() async {
-    const graphQLDocument = r'''mutation MyMutation($name: String!) {
-      createBlog(input: {name: $name}) {
-        id
-        name
-        createdAt
-      }
-    }''';
+  // Create a new comment
+  // Attached to the last created blog and post
+  Future<void> createComment() async {
+    final comment =
+        Comment(content: 'Example Comment - ${uuid()}', post: _post);
+    final req = ModelMutations.create(
+      comment,
+      authorizationMode: _authorizationType,
+    );
 
     final operation = Amplify.API.mutate(
-      request: GraphQLRequest<String>(
-        document: graphQLDocument,
-        variables: <String, dynamic>{'name': 'Test App Blog'},
-        authorizationMode: APIAuthorizationType.userPools,
-      ),
+      request: req,
     );
     _lastOperation = operation;
 
     final response = await operation.response;
+    handleResponse(response);
+  }
+
+  // Cancel the last operation in flight
+  void onCancelPressed() async {
+    try {
+      await _lastOperation?.cancel();
+    } on Exception catch (e) {
+      print('Cancel FAILED');
+      print(e.toString());
+    }
+  }
+
+  // Checks for errors and print the response on screen
+  void handleResponse(
+    GraphQLResponse<Model> response,
+  ) {
     final data = response.data;
     if (data == null) {
       print('errors: ${response.errors}');
+      setState(() {
+        _result = response.errors.toString();
+      });
       return;
     }
 
-    print('Result data: $data');
+    final prettyData = prettifyData(data);
+
     setState(() {
-      _result = data;
+      _result = prettyData;
     });
   }
 
-  void onCancelPressed() async {
-    try {
-      await _lastOperation?.cancel();
-    } on Exception catch (e) {
-      print('Cancel FAILED');
-      print(e.toString());
+  // Checks for errors and print the response on screen
+  void handleSubscriptionEvents(GraphQLResponse<dynamic> event) {
+    if (event.hasErrors) {
+      print('Error(s) received from subscription: ${event.errors}');
+      return;
     }
+    // final prettyData = prettifyData(event.data);
+    final data = 'Subscription event data received: ${event.data}';
+    print(data);
+    setState(() {
+      _result = data;
+    });
+  }
+
+  // Pretty prints the data responses
+  String prettifyData(Model data) {
+    final dataJson = const JsonEncoder.withIndent('  ').convert(data.toJson());
+    print('Result data: $dataJson');
+    return dataJson;
   }
 
   @override
   Widget build(BuildContext context) {
     return ListView(
       padding: const EdgeInsets.all(5),
       children: <Widget>[
+        Column(
+          children: [
+            const SizedBox(
+              width: double.infinity,
```

Comment:
Instead of this you can set `crossAxisAlignment: CrossAxisAlignment.stretch` on the `Column`